### PR TITLE
issue: 1176937 Add delayed ack control

### PR DESCRIPTION
**Pull request by igor-ivanov**
Added possibility to control a logic of sending ACK on received data.
In current implementation ack.quick field operates as on/off only and might be extended to be used in a manner as Linux TCP kernel stack does.

### CODE REVIEW

File: src/vma/util/sys_vars.cpp [hunk 1]
```diff
@@ -1025,6 +1025,10 @@ void mce_sys_var::get_env_params()
 		tcp_nodelay = atoi(env_ptr) ? true : false;
 	}
 
+	if ((env_ptr = getenv(SYS_VAR_TCP_QUICKACK)) != NULL) {
+		tcp_quickack = atoi(env_ptr) ? true : false;
+	}
+
```
Comment (liranoz12) [inline src/vma/util/sys_vars.cpp]: Missing assignment to the default value at the beginning of the function

File: README.txt [hunk 2]
```diff
@@ -740,6 +740,15 @@ Use value of 0 to disable.
...
 
+VMA_TCP_QUICKACK
+If set, disable delayed ack ability.
+This means that TCP responds after every packet.
+For more information on TCP_QUICKACK flag refer to tcp manual page.
+Valid Values are:
+Use value of 0 to disable.
+Use value of 1 for enable.
+Default value is Disabled.
+
```
Comment (liranoz12) [inline README.txt]: Please also update the parameters output and update date

File: src/vma/lwip/tcp.h [hunk 3]
```diff
@@ -418,6 +418,12 @@ struct tcp_pcb {
   u8_t accepts_pending;
 #endif /* TCP_LISTEN_BACKLOG */
 #endif /* VMA_NO_TCP_PCB_LISTEN_STRUCT */
+
+  /* Delayed ACK control */
+  struct {
+    /* Number of quick acks */
+    u8_t quick;
+  } ack;
```
Comment (liranoz12) [inline src/vma/lwip/tcp.h]: Why struct of one element ?

**Comment by rafiw** [comment]
bot:retest

File: src/vma/util/sys_vars.cpp [hunk 4]
```diff
@@ -1025,6 +1025,10 @@ void mce_sys_var::get_env_params()
 		tcp_nodelay = atoi(env_ptr) ? true : false;
 	}
 
+	if ((env_ptr = getenv(SYS_VAR_TCP_QUICKACK)) != NULL) {
+		tcp_quickack = atoi(env_ptr) ? true : false;
```
Comment (liranoz12) [inline src/vma/util/sys_vars.cpp]: Do we want to add this parameter to some the relevant spec ?

**Comment by liranoz12** [comment]
Merged #493

**Pull request closed without merging**